src/components/map/layers/edit/NewRoutePathLayer.tsx

```diff
@@ -147,7 +153,7 @@ class RoutePathLayer extends Component<IRoutePathLayerProps> {
 
         if (
             routePathStore!.routePath &&
-            !this.props.toolbarStore!.isSelected(ToolbarTool.AddNewRoutePath)) {
+            this.props.toolbarStore!.selectedTool === undefined) {
```

Comment:
Do you mean if we could write (!this.props.toolbarStore!.selectedTool)?
It would cause a bit of unwanted behaviour, since selectedTool is an enum, and enum is just basically numbers. If selectedTool happens to be an enum, which is translated to the number 0, then we would get a negative response, even if there is a selectedTool. I therefore though undefined would be more secure.

Reply:
Makes sense